### PR TITLE
Minimal unit test for existing overdwell mechanism #6373

### PR DESCRIPTION
**Pull request by kifir23917**

### CODE REVIEW

File: unit_tests/tests/trigger/test_trigger_decoder.cpp [hunk 1]
```diff
@@ -1192,3 +1192,69 @@ TEST(big, testAssertWeAreNotMissingASpark299) {
...
+
+	eth.smartFireFall(20);
+	ASSERT_EQ( 1,  engine->triggerCentral.triggerState.currentCycle.current_index) << "ci#3";
+
+	eth.smartFireRise(20);
+
+	eth.smartFireFall(20);
+	ASSERT_EQ( 1,  eth.engine.triggerCentral.triggerState.currentCycle.current_index) << "ci#5";
+
+	ASSERT_EQ(3000, Sensor::getOrZero(SensorType::Rpm));
```
Comment (rusefillc) [inline unit_tests/tests/trigger/test_trigger_decoder.cpp:1224]: Your call if you want to use eth.assertRpm(3000); just know that it exists

File: unit_tests/tests/trigger/test_trigger_decoder.cpp [hunk 2]
```diff
@@ -1192,3 +1192,69 @@ TEST(big, testAssertWeAreNotMissingASpark299) {
...
+	// Magic above is the simpliest way to schedule overdwell sparkDown that I found in
+	// testAssertWeAreNotMissingASpark299. May be later we will be able to simplify this spell.
+
+	const efitimeus_t expectedSparkUpTimestampUs = NT2US(17727777); // magic constant from debugger
+
+	const efitimeus_t sparkDownOverdwellDurationUs = MS2US(1.5f * engine->ignitionState.sparkDwell);
+	EXPECT_EQ(4500, sparkDownOverdwellDurationUs);
+
+	const efitimeus_t expectedSparkDownOverdwellTimestampUs = expectedSparkUpTimestampUs + sparkDownOverdwellDurationUs;
+	EXPECT_EQ(expectedSparkDownOverdwellTimestampUs, NT2US(18177776));  // magic constant from debugger
```
Comment (rusefillc) [inline unit_tests/tests/trigger/test_trigger_decoder.cpp:1245]: please attempt magic eth.angleToTimeUs at least in one spot let's see if we recognize the angle value

**Comment by rusefillc** [comment]
@kifir23917 let's 

```
	engine->onIgnitionEvent = [&](IgnitionEvent* event, bool state) {
	    efitick_t startTime = coilStartTimes[event->coilIndex];
```

File: unit_tests/tests/trigger/test_trigger_decoder.cpp (hunk 1 above)
Comment (kifir23917) [reply unit_tests/tests/trigger/test_trigger_decoder.cpp:1224]: I would suggest to get rid of such wrappers at all :)
See #6477

**Pull request merged**